packages/react/dialog/src/Dialog.tsx

```diff
@@ -164,19 +175,112 @@ const DialogContent = React.forwardRef((props, forwardedRef) => {
   const context = useDialogContext(CONTENT_NAME);
   return (
     <Presence present={forceMount || context.open}>
-      <DialogContentImpl data-state={getState(context.open)} {...contentProps} ref={forwardedRef} />
+      {context.modal ? (
+        <DialogContentModalImpl {...contentProps} ref={forwardedRef} />
+      ) : (
+        <DialogContentNonModalImpl {...contentProps} ref={forwardedRef} />
+      )}
     </Presence>
   );
 }) as DialogContentPrimitive;
 
+DialogContent.displayName = CONTENT_NAME;
+
+type DialogContentImplOwnProps = Omit<
+  Polymorphic.OwnProps<typeof DialogContentPart>,
+  'trapFocus' | 'disableOutsidePointerEvents'
+>;
+
+type DialogContentImplPrimitive = Polymorphic.ForwardRefComponent<
+  Polymorphic.IntrinsicElement<typeof DialogContentPart>,
+  DialogContentImplOwnProps
+>;
+
+const DialogContentModalImpl = React.forwardRef((props, forwardedRef) => {
+  const context = useDialogContext(CONTENT_NAME);
+  const contentRef = React.useRef<HTMLDivElement>(null);
+  const composedRefs = useComposedRefs(forwardedRef, contentRef);
+
+  // aria-hide everything except the content (better supported equivalent to setting aria-modal)
+  React.useEffect(() => {
+    const content = contentRef.current;
+    if (content) return hideOthers(content);
+  }, []);
+
+  return (
+    <Portal>
+      <RemoveScroll>
+        <DialogContentPart
```

Comment:
Can we rename this to `DialogContentImpl` for consistency? It looks like you might have copied `Slider` but I've just updated that one too to match everything else. It was the only component that did this "Part" thing, everywhere else we use "Impl".

Reply:
You mean instead of `DialogContentPart`?

Reply:
Haha yeah I think I looked at `Slider` 😀

Reply:
> You mean instead of `DialogContentPart`?
Yep :+1: